### ISSUE TITLE
[no-output-native] False positive when EventEmitter's generic type is an object with a property that is named like a native event

### ISSUE THREAD

**Issue by cypressious** [angular-eslint/angular-eslint#523]
**Description and reproduction of the issue**

The rule `no-output-native` produces a false positive if the generic type of the `EventEmitter` is a complex object type with at least one property that is named like a native event according to https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin/src/rules/no-output-native.ts#L49-L219.

```ts
// The following all produce a false positive

@Output() foo = new EventEmitter<{ click: string }>();
@Output() foo = new EventEmitter<{ blur: string }>();
@Output() foo = new EventEmitter<{ bar: string, blur: string }>();

// The following don't produce a false positive

@Output() foo = new EventEmitter<{ bar: string }>();
@Output() foo = new EventEmitter<{}>();
@Output() foo = new EventEmitter();
@Output() foo = new EventEmitter<ComplextObject>();

type ComplextObject = { click: string };
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `12.1.0` |
| `@typescript-eslint/parser`        | `4.25.0` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.27.0` |
| `node`                             | `v14.17.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

Angular CLI: 12.0.2
Node: 14.17.0
Package Manager: yarn 1.22.5
OS: win32 x64

Angular: 12.0.2
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, localize, material
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.2
@angular-devkit/build-angular   12.0.2
@angular-devkit/core            12.0.2
@angular-devkit/schematics      12.0.2
@schematics/angular             12.0.2
rxjs                            7.1.0
typescript                      4.2.3

```

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.
